### DISCUSSION TITLE
Music plugin disappears after shutting down or stopping Music app

### DISCUSSION THREAD

**Discussion by hamadahiro** [FelixKratz/SketchyBar#443] [Q&A] [answered]
This issue is related to [this](https://github.com/FelixKratz/SketchyBar/discussions/12#discussioncomment-1216811) plugin.

I noticed that this problem was mentioned in the file and a fix was proposed, but it didn't work for me:

```
# FIXME: Running an osascript on an application target opens that app
# This sleep is needed to try and ensure that theres enough time to
# quit the app before the next osascript command is called. I assume 
# com.apple.iTunes.playerInfo fires off an event when the player quits
# so it imediately runs before the process is killed
sleep 1
```

Is there anything I'm missing? Should I add anything else? I have to restart SketchyBar every time I turn on the Music app and play a song, after which it works. I love this plugin and would appreciate any help in finding a solution.

Thanks.

**Comment by FelixKratz** [comment, marked as answer]
Try setting `updates=on` explicitly for this item if you have set `updates=when_shown` in your defaults.

**Comment by hamadahiro** [comment]
I would never have thought of that! Works like a charm. Thanks!

**Reply by FelixKratz** [reply]
I think this item property can cause quite some confusion, because it can bring an item into a state where it can not recover from itself. Setting `updates=when_shown` will disable all script executions, and hence all reactions to events, if the item is not currently drawn to screen, this is a nice performance saving measure. However, if an item sets `drawing=off` in its own script, it can not recover from this, because the script will not be called ever again, as it is no longer drawn to screen. Thus setting `updates=on` forces scripts to be executed even if the item is not drawn to screen currently.

**Reply by hamadahiro** [reply]
I tried setting updates=on for the music.sh item only and reverted the defaults back to when_shown, and it works. Considering all you've said, I think this is a better solution.
